bioblend/_tests/TestGalaxyHistories.py

```diff
@@ -195,5 +195,18 @@ def test_copy_dataset(self):
         self._wait_and_verify_dataset(copied_dataset['id'], expected_contents)
         self.gi.histories.delete_history(self.history_id2, purge=True)
 
+    @test_util.skip_unless_galaxy('release_20.09')
+    def test_update_dataset_datatype(self):
+        history_id = self.history["id"]
+        dataset1_id = self._test_dataset(history_id)
+        original_hda = self.gi.datasets.show_dataset(dataset1_id)
+        self._wait_and_verify_dataset(dataset1_id, b'1\t2\t3\n')
+        assert original_hda['extension'] == 'auto'
+        assert original_hda['data_type'] == 'galaxy.datatypes.data.Data'
+        self.gi.histories.update_dataset(history_id, dataset1_id, datatype='tabular')
+        updated_hda = self.gi.datasets.show_dataset(dataset1_id)
+        assert updated_hda['extension'] == 'tabular'
+        assert updated_hda['data_type'] == 'galaxy.datatypes.tabular.Tabular'
+
```

Comment:
@nsoranzo, I kept it fairly simple here, in comparison to the Galaxy API test. If you would like to also see tests here for all the possible errors that can be raised, let me know.

Reply:
That's fine, thanks!